Moonscraper Chart Editor/Assets/Scripts/Game/UI/Inspectors/NotePropertiesPanelController.cs

```diff
@@ -225,168 +247,72 @@ void Controls()
     {
         if (toggleBlockingActive)
             return;
 
+        Note.Flags newFlags;
         if (IsInNoteTool())
         {
-            SetDoubleKickNoteTool();
+            if (toggle.interactable)
+                newFlags = noteToolController.desiredFlags;
+            else
+                return;
         }
         else
         {
-            SetDoubleKickNote();
-        }
-    }
-
-    void SetForcedNote()
-    {
-        if (currentNote == prevNote)
-        {
-            var newFlags = currentNote.flags;
-
-            if (currentNote != null)
-            {
-                if (forcedToggle.isOn)
-                    newFlags |= Note.Flags.Forced;
-                else
-                    newFlags &= ~Note.Flags.Forced;
-            }
-
-            SetNewFlags(currentNote, newFlags);
-        }
-    }
-
-    void SetForcedNoteTool()
-    {
-        if (forcedToggle.interactable)
-            SetNoteToolFlag(ref noteToolController.desiredFlags, forcedToggle, Note.Flags.Forced);
-    }
-
-    void SetCymbalNote()
-    {
-        if (currentNote == prevNote)
-        {
-            var newFlags = currentNote.flags;
-
-            if (currentNote != null)
-            {
-                if (cymbalToggle.isOn)
-                    newFlags |= Note.Flags.ProDrums_Cymbal;
-                else
-                    newFlags &= ~Note.Flags.ProDrums_Cymbal;
-            }
-
-            SetNewFlags(currentNote, newFlags);
+            if (currentNote == prevNote && currentNote != null)
+                newFlags = currentNote.flags;
+            else
+                return;
         }
 
-    }
-
-    void SetCymbalNoteTool()
-    {
-        if (cymbalToggle.interactable)
-            SetNoteToolFlag(ref noteToolController.desiredFlags, cymbalToggle, Note.Flags.ProDrums_Cymbal);
-    }
-
-    void SetDoubleKickNote()
-    {
-        if (currentNote == prevNote)
+        SetNoteFlag(ref newFlags, flag, toggle.isOn);
+        if ((newFlags & flagToExclude) != Note.Flags.None)
         {
-            var newFlags = currentNote.flags;
-
-            if (currentNote != null)
-            {
-                if (doubleKickToggle.isOn)
-                    newFlags |= Note.Flags.DoubleKick;
-                else
-                    newFlags &= ~Note.Flags.DoubleKick;
-            }
-
-            SetNewFlags(currentNote, newFlags);
+            SetNoteFlag(ref newFlags, flagToExclude, false);
         }
 
+        SetNewFlags(currentNote, newFlags);
     }
 
-    void SetDoubleKickNoteTool()
+    void SetNoteFlag(ref Note.Flags flags, Note.Flags flagsToToggle, bool enable)
     {
-        if (doubleKickToggle.interactable)
-            SetNoteToolFlag(ref noteToolController.desiredFlags, doubleKickToggle, Note.Flags.DoubleKick);
+        if (enable)
+            flags |= flagsToToggle;
+        else
+            flags &= ~flagsToToggle;
     }
```

Comment:
Replace with
```
    Note.Flags EnableFlags(Note.Flags flags, Note.Flags flagsToToggle, bool enabled)
    {
        if (enabled)
            flags |= flagsToToggle;
        else
            flags &= ~flagsToToggle;

        return flags;
    }
```

No point in reusing the SetNoteFlag function name here, makes it very confusing to read.